src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpAddBracesDiagnosticAnalyzer : DiagnosticAnalyzer, IBuiltInAnalyzer
+    {
+        private static readonly LocalizableString s_addBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Add_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_addBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Add_braces_to_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Remove_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Remove_braces_from_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(_noneAddBracesDiagnosticDescriptor,
+                                  _infoAddBracesDiagnosticDescriptor,
+                                  _warningAddBracesDiagnosticDescriptor,
+                                  _errorAddBracesDiagnosticDescriptor,
+                                  _noneRemoveBracesDiagnosticDescriptor,
+                                  _infoRemoveBracesDiagnosticDescriptor,
+                                  _warningRemoveBracesDiagnosticDescriptor,
+                                  _errorRemoveBracesDiagnosticDescriptor);
+
+        private readonly DiagnosticDescriptor _noneAddBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _infoAddBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _warningAddBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _errorAddBracesDiagnosticDescriptor;
+        private readonly Dictionary<DiagnosticSeverity, DiagnosticDescriptor> _severityToAddBracesDescriptorMap;
+
+        private readonly DiagnosticDescriptor _noneRemoveBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _infoRemoveBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _warningRemoveBracesDiagnosticDescriptor;
+        private readonly DiagnosticDescriptor _errorRemoveBracesDiagnosticDescriptor;
+        private readonly Dictionary<DiagnosticSeverity, DiagnosticDescriptor> _severityToRemoveBracesDescriptorMap;
```

Comment:
if static, you can just initialize the values in the initializer.  (including hte dictoinary initializer).